content/A prediction experiment with Machine Learning - Bruno Michetti.md

```diff
@@ -0,0 +1,153 @@
+
+#### Linear Regression
+
+In linear regression, the relationship between the target variable and the predictors is modeled using a linear function:
+
+<img src="images/regression-concept.png" alt="Regression" width="800" />
+
+The main goal is to find the unknown coefficients, which are estimated from the data. That’s why prediction is a supervised learning algorithm. I used a linear regression with two regularization methods: Lasso and Ridge Regression.
+On one hand, the LASSO (Least Absolute Shrinkage and Selection Operator) regression is a process that uses shrinkage. Shrinkage is where data values are shrunk towards a central point, like the mean. Lasso regression performs L1 regularization, which adds a penalty equal to the absolute value of the magnitude of coefficients. Some coefficients can become zero and have been eliminated from the model, something very useful when we have lots of parameters. As I said, I used the [implementation of Lasso from scikit learn](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Lasso.html). With this python class, we can use the fit function to train the model with the preprocessed and scaled dataset, and then use the predict function with new inputs to obtain the prediction of the life expectancy.
+On the other hand, the Ridge regression performs L2 regularization, that is, a penalty using the Euclidean norm. Generally, this algorithm reduces the coefficients but doesn't eliminate them. It’s good to use this type of model when we have overfitting problems. As the Lasso regression, scikit learn provides also a [python class for Ridge regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html), with the fit and predict functions.
+
+#### Decision trees
+
+In regression, a decision tree is a type of model that builds a regression in the form of a tree structure. It breaks down the dataset into smaller subsets while at the same time develops an associated decision tree. In that tree, the leaf nodes are results, and the other nodes are decisions.
+In the experiment, I used the Random Forest model, which builds several decision trees (that is why it is called forest) and then ensembles them. Each tree takes only a random subset of the features and makes the decisions from it. What does “ensembles” mean in machine learning? Well, ensemble methods use multiple learning algorithms to obtain better predictive performance than could be obtained from any of the constituent learning algorithms alone. Random forest it’s also [implemented in scikit learn](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html?highlight=random%20forest%20regressor#sklearn.ensemble.RandomForestRegressor) and has the fit and predict functions.
```

Comment:
> to obtain better predictive performance than could be obtained from any of the constituent learning algorithms alone 

I think it should be:
> than the obtained from...

or

> than the ones that could be obtained from...

What do you think?

Reply:
Ok I'll change it by:
```
to obtain better predictive performance than the one that could be obtained from any of the constituent learning algorithms alone
```